serialization/include/rp_ser.h

```diff
@@ -0,0 +1,361 @@
+ * @param[in] _lock_type Os lock data type. For example in Zephyr struct k_sem can be used.
+ * @param[in] _endpoint_num Endpoint number, used for transport endpoint identify.
+ * @param[in] _endpoint_stack_size Endpoint thread stack size.
+ * @param[in] _endpoint_thread_prio Endpoint thread priority.
+ */
+#define RP_SER_DEFINE(_name, _lock_type, _endpoint_num, _endpoint_stack_size, _endpoint_thread_prio) \
+	__weak extern const struct rp_ser_cmd RP_CONCAT(__start_rp_ser_cmd_decoder_, _name)[];	     \
+	__weak extern const struct rp_ser_cmd RP_CONCAT(__stop_rp_ser_cmd_decoder_, _name)[];	     \
+	__weak extern const struct rp_ser_evt RP_CONCAT(__start_rp_ser_evt_decoder_, _name)[];	     \
+	__weak extern const struct rp_ser_evt RP_CONCAT(__stop_rp_ser_evt_decoder_, _name)[];	     \
+												     \
+	static _lock_type RP_CONCAT(_name, _sem);						     \
+	static const struct rp_ser_endpoint RP_CONCAT(_name, _ep) = {				     \
+		.number = _endpoint_num,							     \
+		.stack_size = _endpoint_stack_size,						     \
+		.prio = _endpoint_thread_prio							     \
+	};											     \
+												     \
+												     \
+	const struct rp_ser_decoders RP_CONCAT(_name, _decoders) = {				     \
+		.cmd_begin = RP_CONCAT(__start_rp_ser_cmd_decoder_, _name),			     \
+		.cmd_end = RP_CONCAT(__stop_rp_ser_cmd_decoder_, _name),			     \
+		.evt_begin = RP_CONCAT(__start_rp_ser_evt_decoder_, _name),			     \
+		.evt_end = RP_CONCAT(__stop_rp_ser_evt_decoder_, _name)				     \
+	};											     \
+												     \
+	static struct rp_ser _name = {			                        	             \
+		.decoders = &RP_CONCAT(_name, _decoders),					     \
+		.ep_conf = &RP_CONCAT(_name, _ep),						     \
+		.rp_sem = (void *)&RP_CONCAT(_name, _sem),					     \
+	}
+
+/**@brief Macro for declaring a serialization instance (not creating it)
+ *
+ * Serialization which are split up over multiple files must have exactly
+ * one file use @ref RP_SER_DEFINE to create module-specific state
+ * and register the decoders data section.
+ *
+ * The other serialization files which could share the same instance should
+ * use this macro instead to creating the new one.
+ *
+ * @param[in] _name Exiting instance name.
+ */
+#define RP_SER_DECLARE(_name) \
+	extern struct rp_ser _name
+
+/**@brief Function for initializing the Remote Procedure Serialization instance.
+ *
+ * This function initialize Remote Procedure Serialization instance, and create a new
+ * tranport endpoint for it. Multiple instance can be initialized during runtime. This ensure
+ * paraller remote function call.
+ *
+ * @param[in] rp Remote procedure instance.
+ *
+ * @retval RP_SUCCESS Initialization was successful.
+ * @retval RP_ERROR_NULL A parameter was NULL.
+ */
+rp_err_t rp_ser_init(struct rp_ser *rp);
+
+/**@brief Function for uninitializing the Remote Procedure Serialization
+ *        instance.
+ *
+ * This function uninitialize the Remote Procedure instance and abort
```

Comment:
Also I think this is much up to RP Trans whether or not to abort a Thread - this will depend on the environment too, bare metal could perhaps just disable a corresponding IRQ.